src/main/kotlin/net/atos/zac/zaken/ZakenService.kt

```diff
@@ -0,0 +1,86 @@
+package net.atos.zac.zaken
+
+import jakarta.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Medewerker
+import net.atos.client.zgw.zrc.model.OrganisatorischeEenheid
+import net.atos.client.zgw.zrc.model.RolMedewerker
+import net.atos.client.zgw.zrc.model.RolOrganisatorischeEenheid
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.client.zgw.ztc.ZTCClientService
+import net.atos.client.zgw.ztc.model.generated.RolType
+import net.atos.zac.identity.model.Group
+import net.atos.zac.identity.model.User
+import net.atos.zac.zoeken.IndexeerService
+import net.atos.zac.zoeken.model.index.ZoekObjectType
+import java.util.UUID
+
+class ZakenService @Inject constructor(
+    val zrcClientService: ZRCClientService,
+    val ztcClientService: ZTCClientService,
+    val indexeerService: IndexeerService
+) {
+    companion object {
+        private val defaultCoroutineScope = CoroutineScope(Dispatchers.Default)
+    }
+
+    /**
+     * Asynchronously assigns a list of zaken to a group and/or user and updates the search index on the fly.
+     */
+    fun assignZakenAsync(zaakUUIDs: List<UUID>, explanation: String, group: Group? = null, user: User? = null) =
+        defaultCoroutineScope.launch {
+            zaakUUIDs.forEach { zaakUUID ->
+                val zaak = zrcClientService.readZaak(zaakUUID)
```

Comment:
Are we going to run co-routine for each zaak in future or the idea is just to detach the long-running function?

Reply:
This is just one coroutine for the entire batch of 'zaken to be assigned and reindexed'. It is not a coroutine per zaak.  The `zaakUUIDs` argument contains the entire list of zaak UUIDs to be assigned.

Reply:
The `launch` call _is_ the detaching of the long running function. Kotlin then runs the entire block of code (the big forEach loop) in a separate coroutine and runs the function call immediately.

In the next PR I will probably switch from `launch` to `async` to be able to return a result after the coroutine has completed to be sent back to the client.

Reply:
Yeah, got this. Do we plan to have parallel rest calls to external systems?

Cause the overall time would still be the same even as async. We'll just move the wait on UI side